### PR TITLE
How to Build an IP Address Tracker with Mapbox API and Vue.

### LINKED ISSUE #5068: [Languages] How to Build an IP Address Tracker with Mapbox API and Vue.
### Writing sample(s): 
- https://egghead.io/blog/how-to-use-the-animate-on-scroll-aos-library-in-vue
- https://www.freecodecamp.org/news/build-accessible-vue-applications/

#### Proposed title of the article
How to Build an IP Address Tracker with Mapbox API and Vue.

#### Proposed article introduction

An IP address is dynamic and can be used for different purposes in an application. With given access to a user's IP address, it's easy to find the location, time zone, and ISP of the user.
In this guide, we’ll build an  IP Address Tracker Application from scratch, using the vue.js framework and the Mapbox API. We’ll cover the process from building the front-end scaffolding up to implementing the API to handle the user's IP address, then go further to visualize the location on the map using the Mapbox API. At the end of this tutorial, readers would also be able to build and deploy the application on Netlify.

#### Key takeaways 
- Understanding of an IP Address.
- The Features of the IP Address Tracker.
- Scaffolding The Front End With Vue.js.
- Integrating the APIs in the Vue App.
- Creating Interactive Map Display with Mapbox API.

#### Article quality
- In-depth guide particularly with Vue.js.
- Step-by-step guidelines for beginners.

#### References
N/A

### PR DESCRIPTION
**Pull request by Abiola-Farounbi**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites? See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #5068

### CODE REVIEW

**Comment by Abiola-Farounbi** [comment]
Hi @vinstex,
I have made the pull request on a new branch.

**Comment by vinstex** [comment]
That was so swift. @hectorkambow, I'd like to review this article.

**Comment by vinstex** [comment]
@Abiola-Farounbi, be sure to join the Slack Eng-Ed channel where we can communicate easily.

**Comment by Abiola-Farounbi** [comment]
I am on the slack channel currently

**Comment by vinstex** [comment]
> I am on the slack channel currently

The eng-ed channel? What is your username?

**Comment by Abiola-Farounbi** [comment]
abiolaEsther

File: content/articles/build-an-ip-address-tracker/index.md [hunk 1]
```diff
@@ -0,0 +1,317 @@
+An IP(Internet Protocol) address is a set of numerical codes that identify where a website or device is located. This is a string of numbers separated by periods. Each number in the set ranges from 0 to 255, that is from 0.0.0.0 to 255.255.255.255. 
+They are of two types which are: IPv4 and IPv6.
+IPv4 is old and limited to a 32-bit address while IPv6 is the new standard and allows for 128-bit addresses. The creation of IPv6 is because all possible IPv4 addresses are close to being assigned, so IPv6 is the new standard.
```
Comment (vinstex) [inline content/articles/build-an-ip-address-tracker/index.md:3]: test comment

**Review by vinstex — CHANGES_REQUESTED** [review]

**Comment by vinstex** [comment]
@Abiola-Farounbi, welcome again to the Eng-Ed community. We are glad to have you here. Nice article. I did my first round of review:

- [x] Total number of words, 750. 
- [x] Flow - It's generally OK. 
- [x] Grammar- This is where I saw a lot of corrections being needed. My advice - Thoroughly proofread your work after you're done with writing. After that, pass it through tools such as [Grammarly](https://www.grammarly.com/) to check for high-level errors or ones which can be easily missed e.g spacing.

A plagiarism check will be done by Hector Kambow or any other any member of the Content moderation team.

Make the requested changes then we can move to the next round. A table of contents comes in handy too, but that we'll add it at the end.

**Review by vinstex — CHANGES_REQUESTED** [review]

**Comment by vinstex** [comment]
@Abiola-Farounbi, commencing tech review. Ammend the minor changes I suggested on Slack.

**Comment by vinstex** [comment]
Hi @Abiola-Farounbi. On doing a tech review, I found out a couple of things to point out.
1. On the first check, your first snippet(<template></template> part) of your *App.vue* is not exact to the one in your GitHub account. Ensure that all your snippets are similar. There are a few codes missing.
2.  Provide code comments as much as possible where necessary, especially the `.vue` files(you can ignore for obvious lines😃).
3.  Ensure that you guide the reader in a **step-by-step** fashion. For example- modify the App.vue file by first adding this code snippet then explaining what the snippet does. Next, add this code- explain. Do that for all the code in a particular file. After that, show the full code for that particular file. I noticed that you only explained a few snippets and did not include all the code. That's why I earlier on requested you to add the full code. Create subheadings for the app and the components. (The reader should comfortably create the end product without looking into your Github code- bear this in mind).
4. For the Tailwind.css setup, you pointed to a link that shows how to do that. However, the *tailwind.config.js* generated is not and can never be similar to yours since there are some changes you make to suit your application design. Guide the reader on modifying the *tailwind.config.js* file with the code you used. Do the same for the *app.css* file. Do that while giving some minor explanations on what the stylings and scripts tend to achieve.
5. For the Vue configuration, let the user know that you will be using the **Vue CLI** version and point to the [documentation](https://cli.vuejs.org/).

> Take your time in doing this for you to bring out a quality article. Feel free to reach me on Slack in case of any difficulty or confusion that may arise.

**Comment by WanjaMIKE** [comment]
There seems to be numerous sentences/paragraphs showing plagiarism as shown in the attached report. Please correct them.

[New Scan 9_47 AM Copyleaks Report.pdf](https://github.com/section-engineering-education/engineering-education/files/7792421/New.Scan.9_47.AM.Copyleaks.Report.pdf)

**Comment by Abiola-Farounbi** [comment]
> Thank you @srishilesh😃.
> 
> Hi @Abiola-Farounbi. On doing a tech review, I found out a couple of things to point out.
> 
> 1. On the first check, your first snippet( part) of your _App.vue_ is not exact to the one in your GitHub account. Ensure that all your snippets are similar. There are a few codes missing.
> 2. Provide code comments as much as possible where necessary, especially the `.vue` files(you can ignore for obvious lines😃).
> 3. Ensure that you guide the reader in a **step-by-step** fashion. For example- modify the App.vue file by first adding this code snippet then explaining what the snippet does. Next, add this code- explain. Do that for all the code in a particular file. After that, show the full code for that particular file. I noticed that you only explained a few snippets and did not include all the code. That's why I earlier on requested you to add the full code. Create subheadings for the app and the components. (The reader should comfortably create the end product without looking into your Github code- bear this in mind).
> 4. For the Tailwind.css setup, you pointed to a link that shows how to do that. However, the _tailwind.config.js_ generated is not and can never be similar to yours since there are some changes you make to suit your application design. Guide the reader on modifying the _tailwind.config.js_ file with the code you used. Do the same for the _app.css_ file. Do that while giving some minor explanations on what the stylings and scripts tend to achieve.
> 5. For the Vue configuration, let the user know that you will be using the **Vue CLI** version and point to the [documentation](https://cli.vuejs.org/).
> 
> > Take your time in doing this for you to bring out a quality article. Feel free to reach me on Slack in case of any difficulty or confusion that may arise.

@vinstex, I sent a message on Slack

**Comment by vinstex** [comment]
Noted.

**Comment by Abiola-Farounbi** [comment]
Hi @vinstex, I have made the edits requested.

**Comment by vinstex** [comment]
> Hi @vinstex, I have made the edits requested.

OK. I'll have a look. Sorry for the late reply.

**Comment by Abiola-Farounbi** [comment]
@vinstex , I made the changes required

**Comment by vinstex** [comment]
> @vinstex , I made the changes required

@Abiola-Farounbi, noted. It's been a pleasure working with you throughout this article- the reliable communications, making requested changes. What an experience! I learnt many new things- I now know where to source a map API for my android projects😃. Fine-tune the changes I told you to work on and implement them in your next articles. A great project you made. I hope to review more of your Eng-Ed articles in the future.

@hectorkambow, this article can be passed through a final review. Where I missed or left out, please let me know. Thank you for granting me this test reviewing opportunity.

@srishilesh, please add a *ready for final review* label for me😃.

**Review by vinstex — APPROVED** [review]

**Comment by lalith1403** [comment]
Jumping on this one 🚀

**Comment by lalith1403** [comment]
Hello @vinstex. Thank you for the detailed review. However, there are several readability issues in the article. The grammatical mistakes and spelling mistakes indicate that the author hasn't gone through the changes requested. Also, it is ideal to double-check the article before marking it ready for final review. I will be closing this PR for now. Feel free to make changes and submit a new PR.

**Comment by vinstex** [comment]
> Hello @vinstex. Thank you for the detailed review. However, there are several readability issues in the article. The grammatical mistakes and spelling mistakes indicate that the author hasn't gone through the changes requested. Also, it is ideal to double-check the article before marking it ready for final review. I will be closing this PR for now. Feel free to make changes and submit a new PR.

Thank you @lalith1403. @Abiola-Farounbi

**Pull request closed without merging**